scripts/bench/compare_branch.sh

```
@@ -0,0 +1,76 @@
+
+if [[ -z "$CURRENT_RESULT" ]]; then
+    echo -e "${RED}No result file found for current branch${NC}"
+    exit 1
+fi
+# Copy to stable name
+cp "$CURRENT_RESULT" "${RESULTS_DIR}/${BENCH}_${CONFIG}_current_${TIMESTAMP}.json"
+CURRENT_RESULT="${RESULTS_DIR}/${BENCH}_${CONFIG}_current_${TIMESTAMP}.json"
+
+# 2. Stash, switch, rebuild, evaluate
+echo -e "${BLUE}--- Switching to branch: ${BRANCH} ---${NC}"
+cd "$REPO_ROOT"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)